File: MyProjects/MAP/Assignment 7/Interpreter/Model/Utilities/IDictionary.cs
using System;
using System.Collections.Generic;
using System.Collections;
using System.Linq;
using System.Text;

namespace Interpreter.Model.Utilities
{
    public interface IDictionary<K, V> : IEnumerable
    {
        void setValue(K key, V value);
        Dictionary<K, V> getMap();
        void delete(K key);
        V getValue(K key);
        void clear();
        void setMap(Dictionary<K, V> m);
        List<V> getValues();
        bool exists(K key);
        List<K> getElements();
    }
}

File: MyProjects/MAP/Practical Exam/Procedures/src/Model/Utilities/IDictionary.java
package Model.Utilities;

import java.util.Collection;
import java.util.List;
import java.util.Map;

public interface IDictionary<K,V> {
    void setValue(K key, V value);
    Map<K,V> getMap();
    void delete(K key);
    V getValue(K key);
    void clear();
    void setMap(Map<K,V> m);
    List<V> getValues();
    List<K> getKeys();
    boolean exists(K key);
    Iterable<K> getElements();
    IDictionary<K, V> copy();
}
